File: Landmarks/Money/MoneyOperation.swift
//
//  Expense.swift
//  Landmarks
//
//  Created by Vitaliy Prikota on 25.12.24..
//

import Foundation

struct MoneyOperation: Identifiable, Codable, Hashable {
	let id: UUID
	var date: Date
	var category: Category
	var amount: Double
	var description: String
	var currency: Currency
	var isExpense: Bool
}

struct MoneyOperationInternal: Identifiable, Codable {
	let id: UUID
	let date: Date
	let category: String
	let amount: Double
	let description: String
	let currency: String
	let isExpense: Bool
}

File: ktor-server/moneyapp/src/main/kotlin/model/MoneyOperation.kt
package com.moneyai.model

import com.moneyai.utils.DateSerializer
import com.moneyai.utils.UUIDSerializer
import kotlinx.datetime.LocalDateTime
import kotlinx.serialization.Serializable
import java.util.*

@Serializable
data class MoneyOperation(
    @Serializable(with = UUIDSerializer::class)
    val id: UUID,
    val date: LocalDateTime,
    val category: String,
    val amount: Double,
    val description: String,
    val currency: String,
    val isExpense: Boolean
)
